scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,276 @@
+    do
+        for output in "${outputs[@]}"
+        do
+            jq --raw-output ".contracts.${contract}.\"${output}\"" expected.json > "expected.${output}.json"
+        done
+
+        assembly=$(cat expected.asm.json)
+        if [ "$assembly" != "" ] && [ "$assembly" != "null" ]
+        then
+            if ! "${SOLC}" --combined-json bin,bin-runtime,opcodes,asm,srcmap,srcmap-runtime --pretty-json --json-indent 4 --import-asm-json expected.asm.json > obtained.json 2> obtained.error
+            then
+                printError
+                printError "$sol_file"
+                cat obtained.error >&2
+                FAILED=$((FAILED + 1))
+                return 0
+            fi
+
+            for output in "${outputs[@]}"
+            do
+                for obtained_contract in $(jq '.contracts | keys | .[]' obtained.json  2> /dev/null)
+                do
+                    jq --raw-output ".contracts.${obtained_contract}.\"${output}\"" obtained.json > "obtained.${output}.json"
+                    if ! diff_files "expected.${output}.json" "obtained.${output}.json"
+                    then
+                        _TESTED=
+                        FAILED=$((FAILED + 1))
+                        return 0
+                    fi
+                done
+            done
+
+            # direct export via --asm-json, if imported with --import-asm-json.
+            if ! "${SOLC}" --asm-json --import-asm-json expected.asm.json --pretty-json --json-indent 4 | tail -n+4 > obtained_direct_import_export.json 2> obtained_direct_import_export.error
+            then
+                printError
+                printError "$sol_file"
+                cat obtained_direct_import_export.error >&2
+                FAILED=$((FAILED + 1))
+                return 0
+            fi
+
+            # reformat jsons using jq.
+            jq . expected.asm.json > expected.asm.json.pretty
+            jq . obtained_direct_import_export.json > obtained_direct_import_export.json.pretty
+            if ! diff_files expected.asm.json.pretty obtained_direct_import_export.json.pretty
+            then
+                _TESTED=
+                FAILED=$((FAILED + 1))
+                return 0
+            fi
+        fi
+    done
+
+    if [ -n "${_TESTED}" ]
+    then
+        TESTED=$((TESTED + 1))
+    fi
+}
+
+# function tests whether exporting and importing again leaves the JSON ast unchanged
+# Results are recorded by adding to FAILED or UNCOMPILABLE.
+# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function testImportExportEquivalence {
+    local sol_file="$1"
+    local input_files="$2"
+    if $SOLC --bin "${input_files}" > /dev/null 2>&1
+    then
+        ! [[ -e stderr.txt ]] || { fail "stderr.txt already exists. Refusing to overwrite."; }
```

Comment:
You only need the `{}` here if you have multiple commands.
```suggestion
        ! [[ -e stderr.txt ]] || fail "stderr.txt already exists. Refusing to overwrite."
```